src/main/java/Main.java

```diff
@@ -0,0 +1,207 @@
+                        newPageValue = maxPageNum;
+                        if (othersCars.size() % ITEM_TOTAL > 0){
+                            newPageValue++;
+                            viewPrevious = true;
+                        }
+                    }
+                    page = newPageValue;
+                }
+                int maxIndex = (page * ITEM_TOTAL - 1), minIndex = (maxIndex - ITEM_TOTAL + 1);
+                for (int i = minIndex; i <= maxIndex; i++) {
+                    if (i < othersCars.size()) {
+                        carsPage.add(othersCars.get(i));
+                    }
+                }
+            }
+
+            templateData.put("currentPage", carsPage);
+            templateData.put("userName", userName);
+            templateData.put("userCars", userCars);
+            templateData.put("pageId", page);
+            templateData.put("previous", previousPage);
+            templateData.put("next", nextPage);
+            templateData.put("viewNext", viewNext);
+            templateData.put("viewPrevious", viewPrevious);
+
+
+            return new ModelAndView(templateData, "index.html");
+        }, new MustacheTemplateEngine());
+
+        Spark.get("/edit" , ((request, response) -> {
+            Map templateData = new HashMap();
+            String carIdString = request.queryParams("carId");
+            int carId = Integer.valueOf(carIdString);
+
+            for (Car car : carsListOfAllUsers){
+                if (car.getCarId() == carId){
+                    templateData.put("editCar", car);
+                }
+            }
+
+            return new ModelAndView(templateData, "edit.html");
+        }));
+
+        /**
+         * Posts to the browser after an action "/login". User creation.
+         */
+        Spark.post("/login", (request, response) -> {
+
+            String userName = request.queryParams("userName");
+            User user = users.get(userName);
+
+            if (user == null){
+                user = new User(userName);
+            }
+
+            Session session = request.session();
+            session.attribute(SESSION_USERNAME, userName);
+
+            response.redirect("/");
+
+            return "";
+        });
+
+        /**
+         * Ends the current user's session.
+         */
+        Spark.post("/logout", (request, response) -> {
+
+            request.session().invalidate();
+            response.redirect("/");
+
+            return "";
+        });
+
+        Spark.post("/create-car", (request, response) -> {
+
+            Session session = request.session();
+            String name = session.attribute(SESSION_USERNAME);
+            User user = users.get(name);
+
+            int year = Integer.valueOf(request.queryParams("year"));
+            String make = request.queryParams("make");
+            String model = request.queryParams("model");
+            String drivetrain = request.queryParams("drivetrain");
+
+            //String url = "/";
```

Comment:
Commented code should removed before turning in the code.  Remember this code is visible to everyone and leaving this in is just sloppy looking.